samza-sql/src/main/java/org/apache/samza/sql/runner/SamzaSqlApplicationRunner.java

```diff
@@ -0,0 +1,90 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.apache.samza.sql.runner;
+
+import org.apache.commons.lang3.Validate;
+import org.apache.samza.application.StreamApplication;
+import org.apache.samza.config.Config;
+import org.apache.samza.config.ConfigRewriter;
+import org.apache.samza.job.ApplicationStatus;
+import org.apache.samza.runtime.AbstractApplicationRunner;
+import org.apache.samza.runtime.LocalApplicationRunner;
+import org.apache.samza.runtime.RemoteApplicationRunner;
+
+
+/**
+ * Application runner implementation for SamzaSqlApplication.
+ * SamzaSqlApplication needs SamzaSqlConfigRewriter to infer some of the configs from SQL statements.
+ * Since Samza's config rewriting capability is available only in the RemoteApplicationRunner.
+ * This runner invokes the SamzaSqlConfig re-writer if it is invoked on a standalone mode (i.e. localRunner == true)
+ * otherwise directly calls the RemoteApplicationRunner which automatically performs the config rewriting .
+ */
+public class SamzaSqlApplicationRunner extends AbstractApplicationRunner {
+
+  private final Config sqlConfig;
+  private final AbstractApplicationRunner appRunner;
+  private final Boolean localRunner;
+
+  public SamzaSqlApplicationRunner(Config config) {
+    this(false, config);
+  }
+
+  public SamzaSqlApplicationRunner(Boolean localRunner, Config config) {
+    super(config);
+    this.localRunner = localRunner;
+    if (localRunner) {
+      // If it is a stand alone runner, Samza doesn't call config re-writer, Hence we need to call it explicitly
```

Comment:
Interesting. Is that a bug in Samza? One of the design goals has been that you generally don't need to care whether your app is being deployed locally or remotely.

Reply:
Based on my chat with Navina, I am not sure whether it is considered a bug. @navina - Thoughts?

Reply:
I think the intention is to remove ConfigRewriter as soon as possible :) It has been discussed many times in the past. @nickpan47  can corroborate to that. 

Honestly, I don't think you should use ConfigRewriters when you are using Samza as a library. The config rewriters can be invoked by the user before passing it into the Runner/StreamApplication. I would consider the same behavior with respect to Samza SQL. So, you should not be using it in the `SamzaSqlApplicationRunner`.

At LinkedIn, the rewriter is used mainly to deal with the multiple kafka cluster expansion in the configs (and exceptional case of using a regex to build the input kafka topic names). But that shouldn't drive our design choices, imo. It is an exception at LinkedIn.

In short, this is not a bug. This is our inability to fully remove the rewriters over the past 2+ releases :)

Reply:
@xinyuiscool @nickpan47 If you see SamzaSQLConfigRewriter you would see that there are some samza configs like ("streams.<streamName>.samza.system") that are inferred from the SQL. What is the recommended approach to add these derived configs to Samza that is uniform across the standalone as well as yarn?